src/mapGenerator/MapWriter.cpp

```diff
@@ -0,0 +1,284 @@
+// Copyright (c) 2005 - 2015 Settlers Freaks (sf-team at siedler25.org)
+//
+// This file is part of Return To The Roots.
+//
+// Return To The Roots is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// Return To The Roots is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Return To The Roots. If not, see <http://www.gnu.org/licenses/>.
+
+#include <iostream>
+#include "mapGenerator/MapWriter.h"
+
+bool MapWriter::Write(const std::string& filePath, Map* map)
```

Comment:
Basically: The map consists of layers. The header is one of them. So you'd want to add all the layers. The map itself is (IRC) an archive of instances of those layers. So you'd need to add the layers in the correct order (IMO simply: map header and some raw data entries) and fill them. Take a look at the map loading from that archive. Full map creating might not be implemented yet, this means it should work, but could be a bit more explicit work than it should